### PR TITLE
User reference documentation for structure extensions

### PR DESCRIPTION
**Pull request by enikao**

### CODE REVIEW

File: docs/reference/structure-extensions.adoc [hunk 1]
```diff
@@ -0,0 +1,137 @@
+:m3: https://lionweb-io.github.io/specification/metametamodel/metametamodel.html
+:mpshelp: https://www.jetbrains.com/help/mps
+:underbracket: &#9141;
+
+= Reference for LionWeb Extensions on MPS' Language Structure Aspect
+:experimental:
+
+LionWeb can represent MPS languages in LionCore.
+It supports most MPS languages without changes to the language.
```
Comment (dslmeinte) [inline docs/reference/structure-extensions.adoc]: “most”: what are the limitations?

File: docs/reference/structure-extensions.adoc [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+== Concepts as partitions
+In LionWeb, link:{m3}#partitions[partitions] serve as the container for one node tree (similar to _models_ in MPS).
+
+Use intention menu:Toggle LionWeb Partition[] to add a boolean attribute `LionWeb partition` to a concept.
+If no such attribute exists, we assume default value `false`.
+
+
+== Optional properties
+LionWeb supports link:{m3}#Feature.optional[optional properties] on structure level, whereas MPS only considers them on link:{mpshelp}/editor.html#propertycell[editor level].
```
Comment (dslmeinte) [inline docs/reference/structure-extensions.adoc:34]: But MPS _does_ have it implicitly on containments and references through the `[0..*]` cardinality, right? So, the exporter could derive optionality from the cardinality. Or did you run into a situation where you (would have) needed to amend the cardinality to get a usable result?

File: docs/reference/structure-extensions.adoc [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+Use intention menu:Toggle LionWeb Optional[] to add a boolean attribute `LionWeb optional` to a property.
+If no such attribute exists, we assume default value `false`, i.e. consider the property as mandatory.
+
+
+== Multi-references
+Unlike MPS, LionWeb supports link:{m3}#Link.multiple[multiple references], i.e. references with cardinality `{underbracket}..n`.
+MPS approximates them with link:{mpshelp}/structure.html#smartreferences[smart references]; LionWeb formalizes this approach.
```
Comment (dslmeinte) [inline docs/reference/structure-extensions.adoc:43]: Good addition!

File: docs/reference/structure-extensions.adoc [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+NOTE: LionWeb-MPS currently does not validate the contents of a `JSON` property.
+This might change in future releases.
+
+
+== Limitations on attributes
+MPS' link:{mpshelp}/structure.html#attributes[attributes] map to LionWeb's link:{m3}#Annotation[Annotations].
+MPS specifies details via _attribute info_.
+
+However, LionWeb does not support all edge cases of attributes:
```
Comment (dslmeinte) [inline docs/reference/structure-extensions.adoc:158]: LionWeb or LionWeb-MPS?

File: docs/reference/structure-extensions.adoc [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+* LionWeb only supports attribute on nodes, i.e. specializations of `NodeAttribute`.
```
Comment (dslmeinte) [inline docs/reference/structure-extensions.adoc]: typo: "attributes"

**Review by dslmeinte — APPROVED** [review]
I only checked the docs, so didn't test the actual thing.

File: docs/reference/structure-extensions.adoc (hunk 2 above)
Comment (enikao) [reply docs/reference/structure-extensions.adoc:34]: For references and containments we of course use the information from MPS; only for properties we have to define our own.

File: docs/reference/structure-extensions.adoc (hunk 1 above)
Comment (enikao) [reply docs/reference/structure-extensions.adoc]: What's listed here (plus one I'll add tomorrow). I'll adjust the text.

**Review by dslmeinte — APPROVED** [review]

**Pull request merged**